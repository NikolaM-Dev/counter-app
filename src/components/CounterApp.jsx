import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value = 10 }) => {
  const [counter, setCounter] = useState(value);

  const handleAdiction = () => setCounter((counter) => counter + 1);

  const handleReset = () => setCounter(value);

  const handleSubtraction = () => setCounter((counter) => counter - 1);

  return (
    <>
      <h1>Counter App</h1>
      <h2>{counter}</h2>
      <button onClick={handleSubtraction}>-1</button>
      <button onClick={handleReset}>Reset</button>
      <button onClick={handleAdiction}>+1</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
};

export default CounterApp;
