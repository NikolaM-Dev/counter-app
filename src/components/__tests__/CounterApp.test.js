import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import CounterApp from '../CounterApp';

describe('Tests in <CounterApp/>', () => {
  let wrapper = shallow(<CounterApp />);

  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });

  it('should to show <CounterApp/> correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should to show the counter value in 100', () => {
    const value = 100;
    const wrapper = shallow(<CounterApp value={value} />);
    const counterText = wrapper.find('h2').text().trim();

    expect(counterText).toBe(String(value));
  });

  it('should increase with button +1', () => {
    wrapper.find('button').at(2).simulate('click');
    const counterText = wrapper.find('h2').text().trim();

    expect(counterText).toBe('11');
  });

  it('should decrease with button -1', () => {
    wrapper.find('button').at(0).simulate('click');
    const counterText = wrapper.find('h2').text().trim();

    expect(counterText).toBe('9');
  });
});
