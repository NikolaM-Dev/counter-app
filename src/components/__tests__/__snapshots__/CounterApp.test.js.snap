// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <CounterApp/> should to show <CounterApp/> correctly 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    +1
  </button>
</Fragment>
`;
